src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/Extensions/HealthCertificate+PDF/HealthCertificate+PDF.swift

```diff
@@ -0,0 +1,331 @@
+
+		try pdfDocument.embedImageAndText(
+			image: qrCodeImage,
+			at: CGPoint(x: 132, y: 436),
+			texts: texts(with: valueSets)
+		)
+
+		return pdfDocument
+	}
+
+	private var pdfTemplate: Data {
+		let templateName: String
+		switch type {
+		case .vaccination:
+			templateName = "VaccinationCertificateTemplate_v4.1"
+		case .test:
+			templateName = "TestCertificateTemplate_v4.1"
+		case .recovery:
+			templateName = "RecoveryCertificateTemplate_v4.1"
+		}
+
+		guard let tempalteURL = Bundle.main.url(forResource: templateName, withExtension: "pdf"),
+			  let templateData = FileManager.default.contents(atPath: tempalteURL.path) else {
+			fatalError("Could not load pdf template.")
+		}
+		return templateData
+	}
+
+	private func texts(with valueSets: SAP_Internal_Dgc_ValueSets) -> [PDFText] {
+		switch entry {
+		case .vaccination(let entry):
+			return vaccinationTexts(for: entry, with: valueSets)
+		case .test(let entry):
+			return testTexts(for: entry, with: valueSets)
+		case .recovery(let entry):
+			return recoveryTexts(for: entry, with: valueSets)
+		}
+	}
+
+	private func vaccinationTexts(for entry: VaccinationEntry, with valueSets: SAP_Internal_Dgc_ValueSets) -> [PDFText] {
+		[
+			PDFText(
+				text: entry.certificateIssuer,
+				color: textColor,
+				font: HealthCertificate.openSansFont,
+				rect: CGRect(x: 19, y: 30, width: 126, height: 23),
+				upsideDown: true
+			),
+			PDFText(
+				text: entry.countryOfVaccination,
+				color: textColor,
+				font: HealthCertificate.openSansFont,
+				rect: CGRect(x: 19, y: 61, width: 126, height: 23),
+				upsideDown: true
+			),
+			PDFText(
+				text: entry.dateOfVaccination,
+				color: textColor,
+				font: HealthCertificate.openSansFont,
+				rect: CGRect(x: 19, y: 93, width: 126, height: 23),
+				upsideDown: true
+			),
+			PDFText(
+				text: entry.formattedValue(for: \VaccinationEntry.marketingAuthorizationHolder, valueSets: valueSets) ?? "",
+				color: textColor,
+				font: HealthCertificate.openSansFont,
+				rect: CGRect(x: 158, y: 173, width: 126, height: 23),
+				upsideDown: true
+			),
+			PDFText(
+				text: entry.formattedValue(for: \VaccinationEntry.vaccineMedicinalProduct, valueSets: valueSets) ?? "",
+				color: textColor,
+				font: HealthCertificate.openSansFont,
+				rect: CGRect(x: 158, y: 230, width: 126, height: 23),
+				upsideDown: true
+			),
+			PDFText(
+				text: entry.formattedValue(for: \VaccinationEntry.vaccineOrProphylaxis, valueSets: valueSets) ?? "",
+				color: textColor,
+				font: HealthCertificate.openSansFont,
+				rect: CGRect(x: 19, y: 303, width: 126, height: 23),
+				upsideDown: true
+			),
+			PDFText(
+				text: entry.formattedValue(for: \VaccinationEntry.diseaseOrAgentTargeted, valueSets: valueSets) ?? "",
+				color: textColor,
+				font: HealthCertificate.openSansFont,
+				rect: CGRect(x: 19, y: 331, width: 126, height: 23),
+				upsideDown: true
+			),
+			PDFText(
+				text: "\(entry.doseNumber)  \(entry.totalSeriesOfDoses)",
```

Comment:
I'm a bit confused, I don't see the "/" but its in the screenshot, whats going on here? 😅

Reply:
It is in the template 🤷‍♂️.